### PR TITLE
Homework: Analyzer program

### PR DESCRIPTION
**Pull request by Owlullaby**
Trello card: 
[Homework: Analyzer program code subsection 18](https://trello.com/c/wUlACjoW/767-2-section-2-homework-analyzer-program-code-subsection-18)

Description:
This is the homework that I have done for Section 2-subsection 18 (Homework: Analyzer program code) where we make use of what we had learned before to write a simple Ruby script (analyzer.rb) that takes in user's inputs and print it onto the console in a predefined format.

I make use of 
1. `get.chomp` to receive input from user
2. string concatenation to concatenate `first_name` and `last_name`
3. `.reverse` to reverse the full name
4. `.length` to count the length of the string `first_name` and `last_name` before adding the integer together to eliminate any spacing in between to be counted

### CODE REVIEW

**Comment by Owlullaby** [comment]
Moved the description to the comment above.

**Comment by tlkiong** [comment]
image

Whats all this?

**Comment by Owlullaby** [comment]
> image
> 
> Whats all this?

I'm not sure. It appeared right after I created PR using `gh pr create`. Those are the previous commit messages.

**Review by tlkiong — CHANGES_REQUESTED** [review]
1. Please think of others who have to review your PR. add proper details on your PR description. Not just open PR and let it be
2. Please think of others who have to review your PR code. Add spacing to make it easier to read your codes

**Comment by Owlullaby** [comment]
> 1. Please think of others who have to review your PR. add proper details on your PR description. Not just open PR and let it be
> 
> 2. Please think of others who have to review your PR code. Add spacing to make it easier to read

I edited the comment above to have description for this PR. Thanks for the reminder.

**Comment by tlkiong** [comment]
Huh? What do you mean?

**Comment by tlkiong** [comment]
Please edit the description of the PR itself. Not comment.

**Comment by Owlullaby** [comment]
> Please edit the description of the PR itself. Not comment.

I tried to find the part that says "description" for PR. Correct me if I'm wrong, it is the first comment?

**Comment by tlkiong** [comment]
image

This is the `description`.
You are right. I never noticed they change it to `comment`. 🤔 
If you are ready for me to review, please hit 
image

Please ensure you address all my concern before asking me to review

**Review by tlkiong — CHANGES_REQUESTED** [review]
You didn't address my point (2). 
Please **always** double check

File: section2-the-ruby-programming-language/strings/analyzer.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+puts "Enter your first name"
+first_name = gets.chomp
+
+puts "Enter your last name"
+last_name = gets.chomp
+
+full_name = first_name + " " + last_name
+reversed_name = full_name.reverse
+name_length = first_name.length + last_name.length
```
Comment (tlkiong) [inline section2-the-ruby-programming-language/strings/analyzer.rb:9]: Shouldn’t full name character length be full_name.length?

**Comment by Owlullaby** [comment]
> You didn't address my point (2). Please **always** double check

Sorry, not sure why I read it as spacing for the description. Added spacing to the code. Thanks for your time.

**Review by tlkiong — CHANGES_REQUESTED** [review]
1. Keep things consistent. Everywhere. 
image
Is it capitalized or not.
2. Check conventional commits on the exact format

File: section2-the-ruby-programming-language/strings/analyzer.rb (hunk 1 above)
Comment (Owlullaby) [reply section2-the-ruby-programming-language/strings/analyzer.rb:9]: If using `full_name.length` the spacing in between `first_name` and `last_name` will be counted. The homework did specify that the spacing in between would not be counted.

Comment (tlkiong) [reply section2-the-ruby-programming-language/strings/analyzer.rb:9]: Ok

**Comment by Owlullaby** [comment]
> 1. Keep things consistent. Everywhere.
>        image
>        Is it capitalized or not.
> 
>     2. Check conventional commits on the exact format

1. I will be more careful next time.
2. Upon checking, it's lowercase for everything.

**Comment by tlkiong** [comment]
Please edit your commit message then :)

**Review by tlkiong — APPROVED** [review]
Very good. Don’t forget to get your buddy to review

**Review by hanyikoh — APPROVED** [review]
I have reviewed all of your comments and codes. So far is good, just one suggestion is maybe next time you can consider add few comments to code for separation, function description or the output of your code so that collaborators easier understand.

**Comment by Owlullaby** [comment]
> I have reviewed all of your comments and codes. So far is good, just one suggestion is maybe next time you can consider add few comments to code for separation, function description or the output of your code so that collaborators easier understand.

Thanks for the suggestions, I will add in next time.

**Pull request merged**